frontend/epfl-infoscience-search/marc_converter.php

```diff
@@ -310,14 +329,24 @@ public static function parse_record($record, $filter_empty=false) {
      * @return the built array
      */
     public static function convert_marc_to_array($marc_xml) {
+        if (!isValidXml($marc_xml)){
+            throw new InfoscienceUnknownContentException();
```

Comment:
Vu que le constructeur d'exception de PHP (https://www.php.net/manual/fr/exception.construct.php) permet de mettre un message, et que tu as 2 endroits où tu peux propager des exceptions ici, est-ce que ça ne serait pas une idée d'ajouter un message qui permettrait d'en savoir un plus ? genre, ici, tu pourrait mettre "XML not valid" ou un truc du style (ou même étendre la fonction `isValidXml` pour qu'elle différencie si c'est "vide", du "html" ou "xml incorrect", je sais pas.
Ceci permettrait de logger l'erreur exact via  `error_log` comme ça on l'a dant `debug.log` si on essaie de comprendre d'où vient le problème.